### DISCUSSION TITLE
Efficient conversion of point cloud model into finite element accurate mesh model

### DISCUSSION THREAD

**Discussion by zhanglepy** [sofa-framework/sofa#5008] [SofaPython3] [answered]
Hello,
By using a depth camera to obtain the surface point cloud model of an object (in ply format), and then stitching it together from multiple perspectives to form a complete point cloud model, what processing is needed to efficiently convert it into a finite element accurate mesh model suitable for SOFA deformation simulation. I previously tried using meshlab software to convert point clouds into models, then using FreeCAD to convert the models into solids, and finally using Gmsh to partition 2D and 3D meshes. However, the grid model established in this way is relatively rough, with a low success rate and a complex process.
Can anyone help?
I look forward to your help！

**Comment by bakpaul** [comment, marked as answer]
Hello ! 

I am not an expert but I can advice a simple solution for a special case : If your object is convex, there is a simple way : use the convex hull of your pointcloud as the surface mesh, then you can either mesh the resulting volume using either CGal or any meshed you like or you can use it with a `SparseGrid` in SOFA to automatically generate a simple sparse grid filling the volume.

If it is not convex, then it is a complex process and your solution might be as complex as any other one. Maybe CGal is able to make something for you, I am not an expert in meshing 3D pointclouds sorry. 

Good luck !

**Comment by zhanglepy** [comment]
Thanks for your reply！
